152_data_reading_program/sensor_reader.py

```diff
@@ -107,16 +110,30 @@ def sensor_loop():
         except RuntimeError as e:
             # Occasionally sensor does not want to respond
             # Ordinarily this should be I2C read error: max entries reached
-            print(e) # Print the error
+            print(e)  # Print the error
             error_count += 1
             if error_count > 10:
                 print("SENSOR FAILURE... killing program")
-                sys.exit() # Kill the program because something bad is happening
+                sys.exit()  # Kill the program because something bad is happening
         # New sensor data is only available once every 2 seconds by default, but here we
         # poll every 0.5 seconds by default, (or 4 times the measurement interval), to try
         # and catch every available update, even if there was a failure to reach the SCD-30.
-        time.sleep(scd.measurement_interval/4)
+        time.sleep(scd.measurement_interval / 4)
+
+
+def get_current_data():
+    """Processes input for the command get_current_data and prints current sensor readings."""
+    command = input()
+    if command.lower() == 'get_current_data':
+        print(f"CO2: {scd.CO2:>6.1f} ppm    "
+              f"T: {scd.temperature:<3.2f}°C    "
+              f"Humidity: {scd.relative_humidity:<3.2f}%")
+    else:
+        print("Expected get_current_data")
+
 
-#Start the sensor
+# Start the sensor
 get_altitude()
-sensor_loop()
+# sensor_loop()
+while True:
+    get_current_data()
```

Comment:
```suggestion
while True:
    command = input()
    if command.lower() == 'get_current_data':
        get_current_data()
    else:
        print(f'{command!r} is not a valid command.')
```

To avoid having to rewrite the command(s) every time, it's also common to print a menu like:
```
1. get current data
2. some other command
3. etc.
```
and ask the user to enter the number.